### PR TITLE
Add documentation for the split env var processor

### PR DESCRIPTION
**Pull request by Orkin**
Documentation for https://github.com/symfony/symfony/pull/31867

### CODE REVIEW

**Comment by ro0NL** [comment]
If this feature is accepted, we should try to clarfiy the diff between `csv` and `split`.

Perhaps advise to use `split` by default, and only `csv` if one needs escaping possibilities.

File: configuration/env_var_processors.rst [hunk 1]
```diff
@@ -380,6 +380,9 @@ Symfony provides the following env var processors:
 
         The ``trim`` processor was introduced in Symfony 4.3.
 
+``env(split:FOO,BAR)``
```
Comment (ro0NL) [inline configuration/env_var_processors.rst]: just `FOO`

File: configuration/env_var_processors.rst [hunk 2]
```diff
@@ -380,6 +380,9 @@ Symfony provides the following env var processors:
...
+    Split the content of ``FOO,BAR`` env var, retrieving a php array.
```
Comment (ro0NL) [inline configuration/env_var_processors.rst]: should mentiond this splits by `,` for now

**Comment by nicolas-grekas** [comment]
Upstream PR has been rejected. Thanks anyway!

**Pull request closed without merging**